modules/create_timepoint/php/create_timepoint.class.inc

```diff
@@ -80,12 +80,23 @@ class Create_Timepoint extends \NDB_Form
             $site = \Site::singleton(intval($values['psc']));
         }
 
+        //Project
+        $projects = \Utility::getProjectList();
+        if (count($projects) == 1) {
+            //if there is only one project, autoselect first project from array of 1
```

Comment:
What happen if `$values['project'] = 14` and the only returned project is `[ 1 => 'LORIS']` ?
Is it ok to auto-select LORIS without notifying the user?

Reply:
I could add a usecase to cover that but then I would throw an exception I guess

Reply:
I think you can just remove the if statement and `\Project::singleton($projects[$project_id]);` will throw a NotFound exception if the project do not exists.

Reply:
if I remove the if it would go straight to `$project_id = intval($values['project']);` then if the `$values['project']` is NULL the `intval()` will return 0 so instead of defaulting to the the only existing project it will default to project=0 which would throw an exception then if project 0 doesn't exist but what it should do is check for any existing project and assign it no ?

Unless i misunderstood which if to remove ?

Reply:
the user should provide valid inputs. 404 Bad Request is a valid response if the project do not exists.

Reply:
If I'm not mistaken, when there is only 1 site or 1 project we dont even display the dropdown so technically the user does not send ANY sites or projects. that behaviour came from studies with single projects or single sites being to lazy to click that one and single option everytime.

given that the front end can submit a site/project or not, the validation expects nothing when only 1 site/project exists but expects a selection when there is more than 1

the process function then defaults to the first site/project from the list of 1 site/project when only one exists or takes the user input

Reply:
i didn't know that the project was optionally provided according to the state of the database.

let's keep the if statement. 

is it also true for site and visit label?

Reply:
site yes, visit label I dont know

Reply:
my recommendation is to move that feature to the frontend but I won't request that change in this PR

Reply:
my recommendation is to move that feature to the frontend but I won't request that change in this PR